SYCL/ESIMD/regression/abs_fix_test.cpp

```diff
@@ -0,0 +1,79 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// TODO online compiler check fails for esimd_emulator
+// XFAIL: esimd_emulator
+//==- abs_fix_test.cpp - Test for abs function -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <ext/intel/esimd.hpp>
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental::esimd;
+
+#define SIMD 16
+#define THREAD_NUM 512
+
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+template <typename T0, typename T1>
+int test_abs(cl::sycl::queue q, T1 test_value) {
+
+  shared_allocator<T1> allocator1(q);
+  shared_vector<T1> A(THREAD_NUM * SIMD, allocator1);
+
+  shared_allocator<T0> allocator0(q);
+  shared_vector<T0> B(THREAD_NUM * SIMD, allocator0);
+  for (int i = 0; i < A.size(); ++i) {
+    A[i] = test_value;
+  }
+
+  nd_range<1> Range((range<1>(THREAD_NUM)), (range<1>(SIMD)));
+  auto e = q.submit([&](handler &cgh) {
+    T1 *in_ptr = A.data();
+    T0 *const out_ptr = B.data();
+    cgh.parallel_for(Range, [=](nd_item<1> it) SYCL_ESIMD_FUNCTION {
+      __ESIMD_NS::simd<T1, SIMD> A_load_vec;
+      A_load_vec.copy_from(in_ptr + it.get_global_id(0) * SIMD);
+
+      __ESIMD_NS::simd<T0, SIMD> result;
+      result = abs<T0, T1, SIMD>(A_load_vec);
+      result.copy_to(out_ptr + it.get_global_id(0) * SIMD);
+    });
+  });
+  e.wait();
+
+  for (int i = 0; i < THREAD_NUM * SIMD; ++i) {
+    if (B[i] != std::abs(A[i])) {
+      std::cout << "Test failed for " << A[i] << "." << std::endl;
+      return 1;
+    }
+  }
+  return 0;
+}
+
+int main(int argc, char *argv[]) {
+  sycl::property_list properties{sycl::property::queue::enable_profiling()};
+  auto q = sycl::queue(properties);
+
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
+
+  int test_result = 0;
+  test_result |= test_abs<uint32_t, int32_t>(q, 0xFFFFFFFF);

```

Comment:
As far as I know, one of customer/reporter's concern was the case when only T1->T0 conversion is defined, this intrinsic is guaranteed to work. And having T0->T1 conversion is not required for abs to work.

Can you add such test where there is only one direction conversion to show that the intrinsic still works?